### ISSUE TITLE
Submission guideline tweaks

### ISSUE THREAD

**Issue by klauer** [pcdshub/pcdsdevices#1002]
## Current behavior

New PR submitters tend to get confused about specific aspects of the process:
* Using pre-commit, or running isort/flake8 separately
* What the sections in the pre-release notes mean and how to fill them in correctly

## Potential fix

For the pre-release notes section, 
* We already have an excellent explanation of the sections:
* https://github.com/pcdshub/pcdsdevices/blob/master/docs/source/upcoming_release_notes/template-full.rst#maintenance
* However, we do not link this from anywhere that I can see or include it in the documentation
* So: include it in the documentation and link it in the PR template seem to be reasonable options

For pre-commit/isort/flake8, we should, perhaps:
* Link existing documentation
* Make an additional checkbox stating that you read those docs?

Other/better ideas?

**Comment by ZLLentz** [comment]
We can also extend the CI to run the pre-commit to help it not fall through the cracks

**Comment by ZLLentz** [comment]
Thoughts on adding a new job for it vs replacing the normal linter with a pre-commit check?

**Comment by tangkong** [comment]
I remember hearing about some repos applying things like isort and flake8 fixes on commit, rather than relying on the user to apply those changes.  

I don't necessarily like that approach, but it does take the problem partially out of the hands of the submitter.

Optimistically, at some point we'll run out of people to teach about pre-commit hooks.  Realistically I think making a page that explains each step and linking it in the issue / PR template would be helpful.

**Comment by klauer** [comment]
> Thoughts on adding a new job for it vs replacing the normal linter with a pre-commit check?

Given that a flake8 check is part of pre-commit, I don't see the need for both. I'd vote on replacing it. It might also make it more obvious that we specifically want submitters to follow the pre-commit steps.

> I remember hearing about some repos applying things like isort and flake8 fixes on commit, rather than relying on the user to apply those changes.

I think that - despite a bit of frustration on the submitter end and our end for enforcing such things - there is potential for people to learn more about good Python development practices. Personally, I'm OK with continuing to nudge people in that direction, even if it is manually.

> Realistically I think making a page that explains each step and linking it in the issue / PR template would be helpful.

Is [this](https://confluence.slac.stanford.edu/display/PCDS/How+to+add+a+device+to+pcdsdevices#Howtoaddadevicetopcdsdevices-DoIreallyneedtomakeanewdevice?) too complicated/too wordy? I'm open to all input there

**Comment by ZLLentz** [comment]
That linked doc is exactly the thing I wished I knew we had that I could send people when they asked about making new devices

**Comment by tangkong** [comment]
I could see some value in a stripped down version that helps people from the point at which they're submitting the PR.  It may be recency bias, but it seems like people are getting their classes set up well enough and stumbling once they hit the PR checklist.  

That doc is definitely something that should be read when a new device is being created, but maybe a short .md file might be helpful?

**Comment by ZLLentz** [comment]
https://github.com/pcdshub/pcds-ci-helpers/pull/72 created to test pre-commit CI

**Comment by ZLLentz** [comment]
I spent way too much time writing this comment so I think it should be re-used in a docs page somewhere on this topic:
https://github.com/pcdshub/pcdsdevices/pull/1001#issuecomment-1110297347

**Comment by tangkong** [comment]
I propose submission guidelines be hosted somewhere on https://pcdshub.github.io/.  I mostly don't want these guidelines to be put behind the confluence login-wall.  

I think a section under `Development` would make sense, since we already cover some basics of github etc.  

Naming conventions might be something that goes into the pcdsdevices docs, since that's more specific?

**Comment by ZLLentz** [comment]
We can include these per-repo as well so you see the guidelines when you make a PR

**Comment by klauer** [comment]
I'd prefer it to be both on confluence and in the repositories, personally.

Confluence being behind a login is annoying, but it more than balances that out by being an easy spot where we can collaboratively edit, update, and search our documentation.
